platforms/org.eclipse.mita.platform.xdk110/src/org/eclipse/mita/platform/xdk110/buses/I2CGenerator.xtend

```diff
@@ -79,6 +87,18 @@ class I2CGenerator extends AbstractSystemResourceGenerator {
 		static void i2cCallback(I2C_T i2c, struct MCU_I2C_Event_S event) {
 			I2CTransceiver_LoopCallback(&tranceiverStruct, event);
 		}
+		
+		static void swapByteOrder(uint8_t* buffer, size_t objectSize, size_t objectCount) {
```

Comment:
Doesn't such a function already exist somewhere in the SDK?

Reply:
In Serval there are some byte order functions but not for explicit reordering, only for "network to host" and vice versa. No guarantee this actually inverts. Also they are not for arrays and not for types other than `(u)int(16|32)_t`.

Reply:
There are ARM specific instructions implementing this that are disabled for compilers other than RealView, and also only `(u)int(16|32)_t`.